grakn-engine/src/main/java/ai/grakn/engine/session/GraqlSession.java

```diff
@@ -218,7 +218,7 @@ void receiveQuery(Json json) {
                 String queryString = queryStringBuilder.toString();
                 queryStringBuilder = new StringBuilder();
 
-                queries = graph.graql().infer(infer).materialise(materialise).parseList(queryString);
+                queries = graph.graql().infer(infer).materialise(materialise).parseList(queryString).collect(toList());
```

Comment:
Good point! I check the list later to see if the queries are read-only, so I'll have to do something a bit nasty, but it's doable.